jdaviz/configs/imviz/plugins/tools.py

```diff
@@ -134,59 +134,56 @@ class ContrastBias(CheckableTool):
     tool_tip = 'Click and drag to adjust contrast and bias, double-click to reset'
 
     def __init__(self, viewer, **kwargs):
-        self._time_last = 0
         super().__init__(viewer, **kwargs)
 
     def activate(self):
         self.viewer.add_event_callback(self.on_mouse_or_key_event,
                                        events=['dragstart', 'dragmove',
-                                               'dragend', 'dblclick'])
+                                               'dblclick'])
 
     def deactivate(self):
         self.viewer.remove_event_callback(self.on_mouse_or_key_event)
 
-    def on_mouse_or_key_event(self, data):
-        from jdaviz.configs.imviz.helper import get_top_layer_index
+    @debounced(delay_seconds=0.03, method=True)
+    def _dragevent(self, data):
+        event_x = data['pixel']['x']
+        event_y = data['pixel']['y']
 
-        event = data['event']
+        if ((event_x < 0) or (event_x >= self._max_x) or
+                (event_y < 0) or (event_y >= self._max_y)):
+            return
 
-        # Note that we throttle this to 200ms here as changing the contrast
-        # and bias it expensive since it forces the whole image to be redrawn
-        if event == 'dragmove':
-            if (time.time() - self._time_last) <= 0.2:
-                return
+        # Normalize w.r.t. viewer display from 0 to 1
+        # Also see glue/viewers/matplotlib/qt/toolbar_mode.py
+        # bias range 0..1
+        # contrast range 0..4
+        bias = event_x / (self._max_x - 1)
+        contrast = 4 * (1 - (event_y / (self._max_y - 1)))
 
-            event_x = data['pixel']['x']
-            event_y = data['pixel']['y']
-            max_x = self.viewer.shape[1]
-            max_y = self.viewer.shape[0]
+        with delay_callback(self._layer_state, 'bias', 'contrast'):
+            self._layer_state.bias = bias
+            self._layer_state.contrast = contrast
 
-            if ((event_x < 0) or (event_x >= max_x) or
-                    (event_y < 0) or (event_y >= max_y)):
-                return
+    def on_mouse_or_key_event(self, data):
+        from jdaviz.configs.imviz.helper import get_top_layer_index
```

Comment:
What's the reason for this import to be here instead of at the top?

Reply:
I think Kyle eventually wants to get rid of it altogether but I am not 100% sure either w.r.t. import move in this PR.